### PR TITLE
Check if WebNN op is supported on the platform

### PR DESCRIPTION
**Pull request by zesongw**
- Add a temp builder in helper
- Map the ONNX Optype and WebNN op by the map in helper.h
[issues/1140](https://github.com/otcshare/webnn-native/issues/1140)

### CODE REVIEW

File: onnxruntime/core/providers/webnn/builders/helper.cc [hunk 1]
```diff
@@ -68,10 +72,30 @@ std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_v
...
+      {0, "auto"}, {1, "high-performance"}, {2, "low-power"}};
+  std::string device_type_name_ = device_type_name_s[webnn_device_flags_];
+  std::string power_preference_name_ = power_preference_name_s[webnn_power_flags_];
+  thread_local const emscripten::val ml = emscripten::val::global("navigator")["ml"];
+  emscripten::val context_options = emscripten::val::object();
+  context_options.set("deviceType", emscripten::val(device_type_name_));
+  context_options.set("powerPreference", emscripten::val(power_preference_name_));
+  emscripten::val wnn_context_ = ml.call<emscripten::val>("createContextSync", context_options);
+  emscripten::val wnn_builder_ = emscripten::val::global("MLGraphBuilder").new_(wnn_context_);
+
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/helper.cc]: Maybe we can move webnn builder creation to webnn_execution_provider.cc.

File: onnxruntime/core/providers/webnn/builders/helper.cc [hunk 2]
```diff
@@ -68,10 +72,30 @@ std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_v
...
   for (size_t i = 0; i < node_indices.size(); i++) {
     auto node_idx = node_indices[i];
     const auto* node(graph_viewer.GetNode(node_idx));
-    bool supported = IsNodeSupported(*node, graph_viewer, logger);
+    bool supported = false;
+    // First check if platform support the op
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/helper.cc]:
```suggestion
    // Firstly check if platform supports the WebNN op
```

File: onnxruntime/core/providers/webnn/builders/helper.h [hunk 3]
```diff
@@ -22,8 +25,28 @@ bool GetShape(const NodeArg& node_arg, std::vector<int64_t>& shape, const loggin
 bool IsInputSupported(const NodeArg& node_arg, const std::string& parent_name, const logging::Logger& logger);
 
 // Get a list of groups of supported nodes, each group represents a subgraph supported by WebNN EP
-std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_viewer,
-                                                      const logging::Logger& logger);
+std::vector<std::vector<NodeIndex>> GetSupportedNodes(const GraphViewer& graph_viewer, const emscripten::val& wnn_builder_, const logging::Logger& logger);
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/builders/helper.h]: Pls. wrap the last two params into lines.

File: onnxruntime/core/providers/webnn/webnn_execution_provider.cc [hunk 4]
```diff
@@ -35,6 +34,29 @@ WebNNExecutionProvider::WebNNExecutionProvider(uint32_t webnn_device_flags, uint
       });
 
   InsertAllocator(CreateAllocator(cpu_memory_info));
+
+    // Create WebNN context and graph builder
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/webnn_execution_provider.cc]: Inconsistent indentation.

File: onnxruntime/core/providers/webnn/webnn_execution_provider.cc [hunk 5]
```diff
@@ -35,6 +34,29 @@ WebNNExecutionProvider::WebNNExecutionProvider(uint32_t webnn_device_flags, uint
...
+  std::unordered_map<uint32_t, std::string> device_type_name_s = {
+    {0, "auto"}, {1, "gpu"}, {2, "cpu"}};
+  std::unordered_map<uint32_t, std::string> power_preference_name_s = {
+      {0, "auto"}, {1, "high-performance"}, {2, "low-power"}};
```
Comment (Honry) [inline onnxruntime/core/providers/webnn/webnn_execution_provider.cc:42]: Ditto.

**Review by Honry — APPROVED** [review]
Generally LGTM, leave a few nits.

**Comment by Honry** [comment]
@huningxin, would you like to take another look?

**Review by huningxin — APPROVED** [review]
LGTM, thanks for the PR @zesongw and the review @Honry !

**Pull request merged**